test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart

```diff
@@ -11,127 +11,186 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-import 'dart:html';
+import 'dart:js';
 
 import 'package:over_react/over_react.dart';
-import 'package:over_react/react_dom.dart' as react_dom;
+import 'package:react/react_client/react_interop.dart' show PropTypes;
 import 'package:test/test.dart';
 
 import '../../../../test_util/test_util.dart';
 
 part 'constant_required_accessor_integration_test.over_react.g.dart';
 
 void main() {
-  group('(Component2) properly identifies required props by', () {
+  group('(Component2) propTypes properly identifies required props by', () {
+    bool consoleErrorCalled;
+    var consoleErrorMessage;
+    JsFunction originalConsoleError;
+
+    setUp(() {
+      consoleErrorCalled = false;
+      consoleErrorMessage = null;
+
+      // PropTypes by default will only throw a specific error one time per Component Class.
+      // This resets the cache after each test so it throws again.
+      // See: https://www.npmjs.com/package/prop-types#proptypesresetwarningcache
+      PropTypes.resetWarningCache();
+
+      originalConsoleError = context['console']['error'];
+      context['console']['error'] = new JsFunction.withThis((self, message, arg1, arg2, arg3) {
+        consoleErrorCalled = true;
+        consoleErrorMessage = message;
+        originalConsoleError.apply([message], thisArg: self);
+      });
+    });
+
+    tearDown(() {
+      context['console']['error'] = originalConsoleError;
+    });
     group('throwing when a prop is required and not set', () {
+
       test('on mount', () {
-        expect(() => render(ComponentTest()..nullable = true),
-            throwsPropError_Required('ComponentTestProps.required')
+        mount(
+          (ComponentTest()..nullable = true)(),
+          attachedToDocument: true,
         );
+
+        expect(consoleErrorCalled, isTrue, reason: 'should have outputted a warning');
+        expect(consoleErrorMessage, contains('ComponentTestProps.required'));
       });
 
       test('on re-render', () {
-        var mountNode = new DivElement();
-        react_dom.render((ComponentTest()
-          ..required = true
-          ..nullable = true
-        )(), mountNode);
+        var jacket = mount((ComponentTest()
+            ..required = true
+            ..nullable = true
+          )(),
+          attachedToDocument: true,
+        );
 
-        expect(() => react_dom.render((ComponentTest()..nullable = true)(), mountNode),
-            throwsPropError_Required('ComponentTestProps.required')
+        expect(consoleErrorCalled, isFalse, reason: 'should not have outputted a warning but found: $consoleErrorMessage');
```

Comment:
Just an idea, another way you could do this to simplify your `expect`s, while still giving you high-quality error messages:
```dart
consoleErrors = [];
context['console']['error'] = new JsFunction.withThis((self, message, arg1, arg2, arg3) {
  consoleErrors.add(message);
  originalConsoleError.apply([message], thisArg: self);
});
```

```diff
- expect(consoleErrorCalled, isFalse, reason: 'should not have outputted a warning but found: $consoleErrorMessage');
+expect(consoleErrors, isEmpty, reason: 'should not have outputted a warning');

...

-expect(consoleErrorCalled, isTrue, reason: 'should have outputted a warning');
-expect(consoleErrorMessage, contains('ComponentTestProps.required'));
+expect(consoleErrorMessages, [contains('ComponentTestProps.required')]);
```